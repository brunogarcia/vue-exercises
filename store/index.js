
import Vuex from 'vuex';

const store = () => {
    return new Vuex.Store({
        state: {
            missionText: '',
            hustonOpen: false,
        },
        getters: {
            hustonText(state) {
                return state.missionText;
            },
            hustonOpen(state) {
                return state.hustonOpen;
            }
        },
        mutations: {
            SET_HUSTON_TEXT(state, text) {
                state.missionText = text;
            },
            CALL_TO_HUSTON(state, val) {
                state.hustonOpen = !!val;
            },
        },
        actions: {
            updateHustonText({ commit }, text) {
                commit('SET_HUSTON_TEXT', text);
            },
            callToHuston({ commit }, val) {
                commit('CALL_TO_HUSTON', val);
            }
        }
    });
};

export default store;
