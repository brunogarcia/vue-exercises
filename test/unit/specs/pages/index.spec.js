import {
    mount,
    createLocalVue
} from '@vue/test-utils';
import Vuex from 'vuex';
import Vuetify from 'vuetify';
import Home from '~/pages/index.vue';
import store from '../../../fixtures/store';

describe('Home /', () => {
    let localVue;

    beforeEach(() => {
        localVue = createLocalVue();
        localVue.use(Vuetify);
        localVue.use(Vuex);
    });

    it('renders home page', () => {
        const wrapper = mount(Home, {
            localVue: localVue,
            store: store(),
            stubs: ['nuxt-link']
        });
        expect(wrapper.isVueInstance()).toBeTruthy();
        expect(wrapper.vm.$el).toMatchSnapshot();
    });
});
