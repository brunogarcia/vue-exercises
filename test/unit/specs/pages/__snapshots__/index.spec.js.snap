// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home / renders home page 1`] = `
<section
  class="container white"
>
  <div
    class="mb-3"
  >
    <img
      alt="Jumbo image"
      class="jumbo-image"
      src="images/cover.jpg"
    />
  </div>
   
  <div
    class="pl-3"
  >
    <h3
      class="subheading"
    >
      Repo: 
      <a
        href="https://github.com/gangsthub/vue-exercises"
        rel="noopener noreferer"
        target="_blank"
      >
        
            https://github.com/gangsthub/vue-exercises
        
      </a>
    </h3>
     
    <p
      class="subheading mt-4"
    >
      
            1️⃣ Go to the first exercise:
            
      <nuxt-link-stub
        class="ibm subheading c-p primary--text"
        tag="span"
        to="[object Object]"
      />
    </p>
     
    <p
      class="subheading"
    >
      
            2️⃣ Remember that you have the "help button"
            
      <button
        class="btn btn--icon primary"
        data-ripple="true"
        slot="activator"
        type="button"
      >
        <div
          class="btn__content"
        >
          <i
            aria-hidden="true"
            class="material-icons icon"
          >
            live_help
          </i>
        </div>
      </button>
      
            available on the
            top right corner for every exercise.
        
    </p>
     
    <p
      class="subheading"
    >
      
            ️3️⃣ Enjoy!
        
    </p>
  </div>
</section>
`;
